File: tutorial-for-assignment1/python/count.py
import sys

filename = sys.argv[1]

f = open(filename, "r")

# map from words to integer count
counts = {}

for line in f.readlines():
    words = line.split()
    for w in words:
        counts[w] = 1 if not w in counts else counts[w]+1

ranked = sorted(counts.keys(), key=lambda x: counts[x], reverse=True)

print "There are %d distinct words" % len(counts)

for w in ranked[:100]:
    print '"%s" %d' % (w, counts[w])

File: tutorial-for-assignment1/java/Count.java
import java.io.*;
import java.util.*;

public class Count {
    public static void main(String[] args) throws Exception {
        String filename = args[0];
        Map<String, Integer> counts = new TreeMap<String, Integer>();

        BufferedReader reader = new BufferedReader(
            new InputStreamReader(
                new FileInputStream(filename)));

        String line = reader.readLine();
        while(line != null) {
            String[] words = line.split("\\s+");
            for(String w : words) {
                if(counts.containsKey(w)) {
                    counts.put(w, counts.get(w)+1);
                } else {
                    counts.put(w, 1);
                }
            }
            line = reader.readLine();
        }
        reader.close();

        System.out.printf("There are %d distinct words.\n", counts.size());
    }
}
